### PR TITLE
chore: use `macos-13` as `macos-12` is deprecated

### PR DESCRIPTION
**Pull request by ggreif**
See: https://github.com/actions/runner-images/issues/10721

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ jobs:
   tests:
     strategy:
       matrix:
-        os: [ ubuntu-latest, macos-12 ]
+        os: [ ubuntu-latest, macos-12, macos-13 ]
```
Comment (ggreif) [inline .github/workflows/test.yml:10]: `macos-12` will be removed in a follow-up PR, as removing it now renders the CI stuck. Maybe there is a better way, I just don't know...

**Review by luc-blaeser — APPROVED** [review]
Thank you for keeping CI running!

File: .github/workflows/test.yml (hunk 1 above)
Comment (ggreif) [reply .github/workflows/test.yml]: Okay, this works now, IDX changed the required check.

**Review by luc-blaeser — APPROVED** [review]
Thanks!

**Comment by ggreif** [comment]
@Mergifyio refresh

**Pull request merged**